src/sync/impl/sync_client.hpp

```diff
@@ -42,6 +46,24 @@ struct SyncClient {
         client.run();
     }) // Throws
     {
+#if REALM_PLATFORM_APPLE
```

Comment:
It might be cleaner to provide an empty, no-op implementation of the interface for non-Apple platforms so that `SyncClient` doesn't need to care which platforms support detecting changes in reachability.